--- platform-auto.h	2016-03-30 22:15:49.565467542 -0700
+++ platform-auto.h	2016-03-30 22:51:46.516000420 -0700
@@ -175,6 +175,9 @@
 	SERIAL_MULTI \
 	CONSOLE_ARG \
 	PSSERIAL0 \
+	"loadbit_addr=0x100000\0" \
+	"bitstream_image=system.bit.bin\0" \
+	"mmc_loadbit_fat=echo Loading bistream from media to RAM..; && mmcinfo && fatload mmc 0 ${loadbit_addr} ${bitstream_image} && fpga load 0 ${loadbit_addr} ${filesize}\0" \
 	"nc=setenv stdout nc;setenv stdin nc;\0" \
 	"ethaddr=00:0a:35:00:1e:53\0" \
 	"sdboot=echo boot Petalinux; mmcinfo && fatload mmc 0 ${netstart} ${kernel_img} && bootm \0" \
@@ -200,7 +203,7 @@
 	"load_kernel=tftpboot ${clobstart} ${kernel_img}\0" \
 	"update_kernel=setenv img kernel; setenv psize ${kernelsize}; setenv installcmd \"install_kernel\"; run load_kernel ${installcmd}; setenv img; setenv psize; setenv installcmd\0" \
 	"install_kernel=mmcinfo && fatwrite mmc 0 ${clobstart} ${kernel_img} ${filesize}\0" \
-	"cp_kernel2ram=mmcinfo && fatload mmc 0 ${netstart} ${kernel_img}\0" \
+	"cp_kernel2ram=mmcinfo && run mmc_loadbit_fat && fatload mmc 0 ${netstart} ${kernel_img}\0" \
 	"dtb_img=system.dtb\0" \
 	"load_dtb=tftpboot ${clobstart} ${dtb_img}\0" \
 	"update_dtb=setenv img dtb; setenv psize ${dtbsize}; setenv installcmd \"install_dtb\"; run load_dtb test_img; setenv img; setenv psize; setenv installcmd\0" \
